src/hooks/useNotificationModal.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  SaveSubscriptionMutation,
+  SaveSubscriptionMutationVariables,
+} from "gql/generated/types";
+import { useBannerDispatchContext } from "context/banners";
+import { useState, useEffect } from "react";
+import { useMutation } from "@apollo/react-hooks";
+import { SAVE_SUBSCRIPTION } from "gql/mutations/save-subscription";
+import get from "lodash.get";
+
+export interface UseNotificationModalProps {
+  subscriptionMethodControls: SubscriptionMethods;
+  triggers: Trigger[];
+  resourceType: ResourceType;
+  resourceId: string;
+  onCancel: () => void;
+}
+export const useNotificationModal = ({
+  triggers,
+  subscriptionMethodControls,
+  resourceType,
+  resourceId,
+  onCancel,
+}: UseNotificationModalProps) => {
+  const dispatchBanner = useBannerDispatchContext();
+  const [selectedSubscriptionMethod, setSelectedSubscriptionMethod] = useState(
+    ""
+  );
+  // target represents the input value for a subscription method
+  const [target, setTarget] = useState<Target>({});
+  const [isFormValid, setIsFormValid] = useState(false);
+  const [extraFieldErrorMessages, setExtraFieldErrorMessages] = useState<
+    string[]
+  >([]);
+  const [saveSubscription, { loading: mutationLoading }] = useMutation<
```

Comment:
Im not sure if this mutation should live within the hook especially since its getting repackaged and exported out. It might make more sense for it to live within the component.

Reply:
I'm not familiar with the patterns for this, but if the mutation will be the same each time does it make sense to keep here? That way the parent component doesn't need to worry about it

Reply:
The hook isnt really doing to much with the mutation its just repackaging it. It will also simplify testing since he wont have to mock the mutation.

Reply:
I'd actually like to keep the mutation in the hook to keep the parent component clean and also because this hook will only ever be used with the NotificationModal component. We don't need to mock the mutation because we test it in e2e tests.

Reply:
i think mutations make sense within hooks too

Reply:
sgtm

Reply:
I moved the mutation the parent component. This made sense because I wanted to test the getRequestPayload function that the hook exports and also because the hook has less contextual dependencies (banners and apollo) which made is simpler to test.